chromium_src/extensions/browser/extension_creator.cc

```diff
@@ -0,0 +1,29 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "extensions/browser/extension_creator.h"
+
+#include "base/command_line.h"
+#include "base/files/file_path.h"
+#include "components/crx_file/crx_creator.h"
+#include "components/crx_file/id_util.h"
+
+namespace {
+const char kPublisherKeySwitch[] = "brave-extension-publisher-key";
+}  // namespace
+
+#define BRAVE_CREATE_CRX(output_path, zip_path, signing_key)        \
+  const auto* cmd = base::CommandLine::ForCurrentProcess();         \
+  std::unique_ptr<crypto::RSAPrivateKey> publisher_key;             \
+  if (cmd->HasSwitch(kPublisherKeySwitch)) {                        \
+    publisher_key =                                                 \
+        ReadInputKey(cmd->GetSwitchValuePath(kPublisherKeySwitch)); \
+    if (!publisher_key)                                             \
+      return false;                                                 \
```

Comment:
We need private `ReadInputKey()`.
Therefore we can convert this to a function only if pass to it `base::BindRepeating(&ExtensionCreator::ReadInputKey, base::Unretained(this))`.
It can't be one line patch => it will be a #define + a function.
Do you think it's worth changing?

Reply:
> Do you think it's worth changing?
missed `ReadInputKey` requirement, so let's keep it as is. but I don't like `return false;`, we should set result and continue instead of returning early.

Reply:
wdyt about `#define ... if(...) { } else` so else would go to `result = crx_file::Create(crx_path, zip_path, private_key);`?

Reply:
Discussed in person: It's ok to always use the brave patched function to catch possible problems earlier if chromium change somethings.